### ISSUE TITLE
Sporadic failure calling ESLint - Error: ENOENT: no such file or directory, stat '[PROJECT_ROOT]/package.json'

### ISSUE THREAD

**Issue by charvey77** [eslint/eslint#13170]
**Tell us about your environment**

* **ESLint Version:** 5.4.0
* **Node Version:** 10.16.2
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "semistandard",
  "env": {
    "amd": true,
    "jasmine": true,
    "qunit": true
  },
  "globals": {
    "F": "readonly"
  },
  "rules": {
    "indent": ["warn", 4, {"SwitchCase": 1}]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// no relevant source code
```

```bash
Node.js/bin/node /build_harness/1.0.2/node_modules/eslint/bin/eslint.js --config /build_harness/1.0.2/node_modules/eslint-config-custom/eslintrc.json [PROJECT_ROOT]/js_src/*.js 
```

**What did you expect to happen?**
ESLint command would parse the specified JavaScript files correctly

**What actually happened? Please include the actual, raw output from ESLint.**

For some background, we have a setup where we use Makefiles to call Node.js to invoke ESLint. We see the following error only **sporadically**. Most of the time, ESLint works as expected. 

What I **think** should be happening is that the package.json that is referenced in the error below should be found here:
**build_harness/1.0.2/node_modules/eslint-config-custom/**

not in:
**[PROJECT_ROOT]**

It feels like some ESLint configuration is getting modified, but I'm not sure how to debug that kind of behavior. Any insights would be most appreciated.

Error: ENOENT: no such file or directory, stat '[PROJECT_ROOT]/package.json'
    at Object.statSync (fs.js:855:3)
    at entries.forEach.entry (/build_harness/1.0.2/node_modules/eslint/lib/util/file-finder.js:49:20)
    at Array.forEach (<anonymous>)
    at normalizeDirectoryEntries (/build_harness/1.0.2/node_modules/eslint/lib/util/file-finder.js:45:13)
    at FileFinder.findAllInDirectoryAndParents (/build_harness/1.0.2/node_modules/eslint/lib/util/file-finder.js:106:30)
    at findAllInDirectoryAndParents.next (<anonymous>)
    at Config.getLocalConfigHierarchy (/build_harness/1.0.2/node_modules/eslint/lib/config.js:220:20)
    at Config.getConfigHierarchy (/build_harness/1.0.2/node_modules/eslint/lib/config.js:192:43)
    at Config.getConfigVector (/build_harness/1.0.2/node_modules/eslint/lib/config.js:299:21)
    at Config.getConfig (/build_harness/1.0.2/node_modules/eslint/lib/config.js:342:29)

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by nzakas** [comment]
Sorry we lost track of this. If you run ESLint with —debug, you might be able to figure out what’s happening.

**Comment by mysticatea** [comment]
Looks like you are using old ESLint. Would you try the latest version? Because ESLint 6.0.0 has changed the configuration loading logic from scratch.

**Comment by charvey77** [comment]
Thanks for the feedback folks! 

I'll need to get a wider approval to update to the latest version, but I'm hoping to be able to get there within the next month or two.

In the meantime, using --debug, I was finally able to get an instance of our sporadic failure. I have attached two log files, both from the same component but from different jobs in our automated system.

At line 54 of the failing log, you can see where it tries to find the package.json at the root of our filer, and not from within our build harness.

At the same point in the passing log, you can see that it apparently finds the correct package.json (still not sure what path it's finding, but it doesn't fail), and ESLint proceeds as expected.

Is there any more debugging I can do at this point in v5.16?

[failure_log.txt](https://github.com/eslint/eslint/files/4560765/failure_log.txt)
[passing_log.txt](https://github.com/eslint/eslint/files/4560766/passing_log.txt)

**Comment by charvey77** [comment]
A quick update: I have gotten approval to update our internal version to 6.8.0. I'm qualifying our changes to see if this clears up the issue I originally reported. I'll will follow up once I have a good sample size of builds using the latest version, and how they respond to the latest ESLint version.

Thanks again for all of the feedback here! Be in touch soon!
